### ISSUE TITLE
DF-MP2 providing wrong energies for larger basis sets

### ISSUE THREAD

**Issue by stevenvdb** [psi4/psi4#408]
I am trying to calculate DF-MP2 counterpoise-corrected energies for some dimers.
PSI4 gives some weird results for aug-cc-pvtz and aug-cc-pvqz (while all seems well for aug-cc-pvdz).

First I provide the results for the benzene-benzene dimer.
![benzene_benzene](https://cloud.githubusercontent.com/assets/4300354/16049263/4eba59cc-3257-11e6-9fb4-7270fba551ed.png)
I plot 4 points of the dissociation curve, show both HF and DF-MP2 interaction energies for the three basis sets mentioned earlier and compare with MOLPRO. Here everything seems to be perfectly fine.

However when I do exactly the same for the boroxine dimer, things look different:
![boroxine_boroxine_rot60](https://cloud.githubusercontent.com/assets/4300354/16049322/92511392-3257-11e6-9e03-9e2922fe86e6.png)
Clearly PSI4 produces some weird results for the DF-MP2 energy in the case of aug-cc-pvtz and aug-cc-pvqz.
I attach the input/output for one case where PSI4 does something weird. (Note that you can skip the CC part to reproduce the problem)

[cbs.in.txt](https://github.com/psi4/psi4/files/314339/cbs.in.txt)
[cbs.out.txt](https://github.com/psi4/psi4/files/314340/cbs.out.txt)

My PSI4 version is Git: Rev {master} e4070d002474257

**Comment by dgasmith** [comment]
Hmm, well thats interesting. Would you mind running the boroxine test case with the DF-MP2 module and post the result? Best way to access this module is to simply request 'energy('MP2')`.

**Comment by kannon92** [comment]
Is it possible that the wrong auxiliary basis set is being used?  I personally never use the automatic df basis sets.

**Comment by loriab** [comment]
I've just checked, and all the orbital and fitting basis sets have boron. Could you run with `set print 3`? That should trigger printing of exactly what basis sets are loaded and from where in lines like

```
   => Loading Basis Set <=

    Role: BASIS
    Keyword: BASIS
    Name: STO-3G
    atoms 1   entry O          line    80 file /home/dsmith/Gits/dgas_psi/share/basis/sto-3g.gbs
    atoms 2-3 entry H          line    18 file /home/dsmith/Gits/dgas_psi/share/basis/sto-3g.gbs
```

**Comment by stevenvdb** [comment]
Here are the results for one example with aug-cc-pvtz, PSI4 output is attached.
```
                  HF           MP2
MOLPRO    -601.62705    -603.59886
PSI4      -601.62713    -603.55035
```
Note that usually differences between MOLPRO and PSI4 are smaller than 1e-3 Hartree, with the remaining difference probably due to differences in df basis sets.

[dfmp2.out.txt](https://github.com/psi4/psi4/files/315714/dfmp2.out.txt)

**Comment by dgasmith** [comment]
For dunning basis sets Psi4 and MOLPRO should be using the same fitting basis set. For 99% of cases the default basis sets will choose the "correct" fitting basis. The difference are still a bit larger than I am comfortable with so id like to track this down. Can you post the MOLPRO output as well?

**Comment by bozkaya** [comment]
DFOCC module is also has DF-MP2 code. Can you run the same job with the DFOCC module DF-MP2 code (You need to set "qc_module occ" option)? By this way we may find out whether the problem is arising from SCF or DFMP2 module.

**Comment by stevenvdb** [comment]
@dgasmith Here's the MOLPRO output
[dfmp2.out.txt](https://github.com/psi4/psi4/files/316840/dfmp2.out.txt)

@bozkaya Here's the output for DFOCC. Interestingly, the canonical DF-MP2 energy is now -603.599 (which is close to the MOLPRO value). The DF-OMP2 energy reported is -603.628
[dfomp2.out.txt](https://github.com/psi4/psi4/files/316848/dfomp2.out.txt)

**Comment by dgasmith** [comment]
Forgot that MOLPRO used non-augmented basis sets always for HF. Switching to `cc-pvtz-jkfit` we get:

```
MOLPRO: -601.627052047
Psi4:   -601.627052093
```

Much better!

Now the MP2 issues comes due to the fact that Psi4 was enforcing symmetry while MOLPRO was not. Running both Psi4 and MOLPRO in C1 symmetry returns the following:

```
MOLPRO:  -603.5988606
Psi4:    -603.5988594
```

So an overall error of `1e-6`, roughly on the order of the convergence thresholds. So there is some kind of symmetry breaking operation going on here. Could be physical or could be a bug. If I recall correctly @bozkaya's code is not "symmetry tolerant" so it forces C1 symmetry from the start. This would explain why DFOCC is returning a result close to MOLPRO, the different SCF fitting basis sets would explain the rest. 

Short version: Its a symmetry issue, specify C1 symmetry and it should fix the errors that you see.

**Comment by stevenvdb** [comment]
Confirmed, specifying C1 symmetry fixes all problems.
